packages/common/src/constructors.ts

```diff
@@ -33,18 +34,21 @@ export function createCustomCommon(
 }
 
 /**
- * Static method to load and set common from a geth genesis JSON
- * @param genesisJSON JSON of geth configuration
+ * Static method to load and set common from a geth genesis object
+ * @param gethGenesis GethGenesis object
  * @param  opts additional {@link GethConfigOpts} for configuring common
  * @returns Common
  */
 export function createCommonFromGethGenesis(
-  genesisJSON: any,
+  genesisJSON: GethGenesis,
   { chain, eips, genesisHash, hardfork, params, customCrypto }: GethConfigOpts,
 ): Common {
   const genesisParams = parseGethGenesis(genesisJSON, chain)
   const common = new Common({
-    chain: genesisParams,
+    chain: {
+      ...genesisParams,
+      name: genesisParams.name ?? 'Custom chain',
+    } as ChainConfig, // Typecasting because of `string` -> `PrefixedHexString` mismatches
```

Comment:
This `PrefixedHexString` type mismatching between different packages is super annoying, we should look if we can fix this at some point. I find myself just typing `PrefixedHexString` as `string` to avoid having to explicitly cast strings which are certainly prefixed.

Reply:
We should do this *before* breaking releases, this will likely lead to Type-incompatibility issues regarding shipped types, and so better do this before.

Reply:
If something is a string that needs to be converted/typecasted as a PrefixedHexString, one can easily just use the `addHexPrefix` which will conditionally add `0x` if not already present. 

Happy to re-open that debate, but do we find that there's no value in knowing for certain that a particular string is a `0x`-prefixed hex string, and not just "some string"? I find it much clearer in usage personally, and it is definitely common good practice at the app-level (with tools like viem that will reject "strings" with types like these: https://viem.sh/docs/glossary/types.html#hash)

I find this similar to the old JS vs TS discussion, where the annoyance of dealing with types is well justified by the ease of debugging and code predictability